### PR TITLE
Migrate Service Bus Administration to azure-xml

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

This PR uses unreleased changes in `azure-xml` and Autorest Java, this cannot be merged until those are merged and released.

This PR migrates `azure-messaging-serivebus`'s administration clients to using `azure-xml` XML serialization and deserialization instead of Jackson Dataformat XML.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core-experimental/src/test/java/com/azure/core/experimental/http/SimpleXmlSerializable.java [hunk 1]
```diff
@@ -30,7 +31,13 @@ public SimpleXmlSerializable(boolean aBooleanAsAttribute, int anInt, double aDec
 
     @Override
     public XmlWriter toXml(XmlWriter xmlWriter) throws XMLStreamException {
-        xmlWriter.writeStartElement("SimpleXml");
+        return toXml(xmlWriter, null);
+    }
+
+    @Override
+    public XmlWriter toXml(XmlWriter xmlWriter, String rootElementName) throws XMLStreamException {
+        rootElementName = CoreUtils.isNullOrEmpty(rootElementName) ? "SimpleXml" : rootElementName;
```
Comment (lmolkova) [inline sdk/core/azure-core-experimental/src/test/java/com/azure/core/experimental/http/SimpleXmlSerializable.java:39]: is `SimpleXml` a magic string?

**Review by lmolkova — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/pom.xml [hunk 2]
```diff
@@ -51,6 +51,11 @@
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-xml</artifactId>
+      <version>1.0.0-beta.2</version> <!-- {x-version-update;unreleased_com.azure:azure-xml;dependency} -->
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/servicebus/azure-messaging-servicebus/pom.xml:57]: Not yet, this PR will stand as a proof of concept more than anything else for now. I will rip out the changes to azure-xml, and Core libraries, from this PR so they can ship this month.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/NamespaceProperties.java [hunk 3]
```diff
@@ -213,4 +195,124 @@ public NamespaceProperties setNamespaceType(NamespaceType namespaceType) {
...
+        return toXml(xmlWriter, null);
+    }
+
+    @Override
+    public XmlWriter toXml(XmlWriter xmlWriter, String rootElementName) throws XMLStreamException {
+        rootElementName = CoreUtils.isNullOrEmpty(rootElementName) ? "NamespaceInfo" : rootElementName;
+        xmlWriter.writeStartElement(rootElementName);
+        xmlWriter.writeNamespace("http://schemas.microsoft.com/netservices/2010/10/servicebus/connect");
+        xmlWriter.writeStringElement(
+                "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect", "Alias", this.alias);
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/NamespaceProperties.java:210]: Yeah, that's next on the list for code generation to make these namespaces constant. I just need to find a good way to name them.

**Pull request closed without merging**